samples/1_Utils/hipDispatchLatency/hipDispatchEnqueueRateMT.cpp

```diff
@@ -0,0 +1,174 @@
+{
+    //resources necessary for this thread
+    hipStream_t stream;
+    hipStreamCreate(&stream);
+    hipModule_t module;
+    hipFunction_t function;
+    hipModuleLoad(&module, "test_kernel.code");
+    hipModuleGetFunction(&function, module, "test");
+    void* kernel_params = nullptr;
+    std::array<float, TOTAL_RUN_COUNT> results;
+
+    //synchronize all threads, before running
+    int tid = shared->fetch_add(1, std::memory_order_release);
+    while (max_threads != shared->load(std::memory_order_acquire)) {}
+
+    for (auto i = 0; i < TOTAL_RUN_COUNT; ++i) {
+        auto start = std::chrono::high_resolution_clock::now();
+        hipModuleLaunchKernel(function, 1, 1, 1, 1, 1, 1, 0, stream, &kernel_params, nullptr);
+        auto stop = std::chrono::high_resolution_clock::now();
+        results[i] = std::chrono::duration<double, std::milli>(stop - start).count();
+    }
+    print_timing("Thread ID : " + std::to_string(tid) + " , " + "hipModuleLaunchKernel enqueue rate", results);
+    hipStreamSynchronize(stream);
+    hipModuleUnload(module);
+    hipStreamDestroy(stream);
+}
+
+// Measure time taken to enqueue a kernel on the GPU using hipLaunchKernelGGL
+void hipLaunchKernelGGL_enqueue_rate(std::atomic_int* shared, int max_threads)
+{
+    //resources necessary for this thread
+    hipStream_t stream;
+    hipStreamCreate(&stream);
+    std::array<float, TOTAL_RUN_COUNT> results;
+
+    //synchronize all threads, before running
+    int tid = shared->fetch_add(1, std::memory_order_release);
+    while (max_threads != shared->load(std::memory_order_acquire)) {}
+
+    for (auto i = 0; i < TOTAL_RUN_COUNT; ++i) {
+        auto start = std::chrono::high_resolution_clock::now();
+        hipLaunchKernelGGL((EmptyKernel), dim3(NUM_GROUPS), dim3(GROUP_SIZE), 0, stream);
+        auto stop = std::chrono::high_resolution_clock::now();
+        results[i] = std::chrono::duration<double, std::milli>(stop - start).count();
+    }
+    print_timing("Thread ID : " + std::to_string(tid) + " , " + "hipLaunchKernelGGL enqueue rate", results);
+    hipStreamSynchronize(stream);
+    hipStreamDestroy(stream);
+}
+
+// Simple thread pool
+struct thread_pool {
+    thread_pool(int total_threads) : max_threads(total_threads) {}
+    void start(std::function<void(std::atomic_int*, int)> f) {
+        for (int i = 0; i < max_threads; ++i) {
+            threads.push_back(std::async(std::launch::async, f, &shared, max_threads));
+        }
+    }
+    void finish() {
+        for (auto&&thread : threads) {
+            thread.get();
+        }
+        threads.clear();
+        shared = {0};
+    }
+    ~thread_pool() {
+        finish();
+    }
+private:
+    std::atomic_int shared {0};
+    std::vector<std::future<void>> threads;
+    int max_threads = 1;
+};
+
+
+int main(int argc, char* argv[])
+{
+    if (argc != 3) {
+        std::cerr << "Run test as 'hipDispatchEnqueueRateMT <num_threads> <0-module_launch /1-ggl_launch /2-both>'\n";
```

Comment:
Worth printing full names since the output of this test doesn't abbreviate. 
0-hipModuleLaunchKernel 
1-hipLaunchKernelGGL

I don't think both should be an option, since: 
a) not common usage
b) later test does not start from scratch, but instead would used HIP resources reserved from previous test.

Reply:
Thanks updated the code.